x-pack/plugins/spaces/server/config.ts

```diff
@@ -25,11 +26,21 @@ export function createConfig$(context: PluginInitializerContext) {
 }
 
 const disabledDeprecation: ConfigDeprecation = (config, fromPath, addDeprecation) => {
-  if (config.xpack?.spaces?.enabled === false) {
+  if ('enabled' in (settings?.xpack?.spaces || {}) {
     addDeprecation({
-      message: `Disabling the Spaces plugin (xpack.spaces.enabled) will not be supported in the next major version (8.0)`,
+      title: i18n.translate('xpack.spaces.deprecations.enabledTitle', {
+        defaultMessage: 'Setting "xpack.spaces.enabled" is deprecated',
+      }),
+      message: i18n.translate('xpack.spaces.deprecations.enabledMessage', {
+        defaultMessage:
+          'In 8.0.0 and later, using "xpack.spaces.enabled" to disable the Spaces plugin is unsupported.',
```

Comment:
Maybe we should also change this message?
How about this?

```suggestion
          'In 8.0.0 and later, using "xpack.spaces.enabled" is unsupported, and the Spaces plugin is always enabled.',
```